test_runner.py

```diff
@@ -273,6 +273,39 @@ def build_test_list():
             "fsdp2_mem_tracker",
             ngpu=4,
         ),
+        OverrideDefinitions(
+            [
+                [
+                    "--training.fp8_linear",
+                ]
+            ],
+            "FSDP2 with bf16 all-gather",
+            "fp8_fsdp2_bf16_all_gather",
+            ngpu=4,
+        ),
+        OverrideDefinitions(
+            [
+                [
+                    "--training.fp8_linear",
+                    "--training.enable_fsdp_fp8_all_gather",
```

Comment:
should it only be enabled when `fp8_linear` and FSDP2 are used?
shall we add an assertion somewhere?

Reply:
good point. so the logic is like follows. if `fp8_linear=False`, `enable_fsdp_fp8_all_gather` won't be executed regardless of true or false. do you prefer to move it to top-level?  

```
if fp8_linear:
   if enable_fsdp_fp8_all_gather
```